async-nats/src/jetstream/types.rs

```diff
@@ -0,0 +1,193 @@
+/// remove older messages. `New` will fail to store the new message.
+#[derive(Debug, Serialize, Deserialize, Clone, Copy, PartialEq, Eq)]
+#[repr(u8)]
+pub enum DiscardPolicy {
+    /// will remove older messages when limits are hit.
+    #[serde(rename = "old")]
+    Old = 0,
+    /// will error on a StoreMsg call when limits are hit
+    #[serde(rename = "new")]
+    New = 1,
+}
+
+impl Default for DiscardPolicy {
+    fn default() -> DiscardPolicy {
+        DiscardPolicy::Old
+    }
+}
+
+/// `RetentionPolicy` determines how messages in a set are retained.
+#[derive(Debug, Serialize, Deserialize, Clone, Copy, PartialEq, Eq)]
+#[repr(u8)]
+pub enum RetentionPolicy {
+    /// `Limits` (default) means that messages are retained until any given limit is reached.
+    /// This could be one of mesages, bytes, or age.
+    #[serde(rename = "limits")]
+    Limits = 0,
+    /// `Interest` specifies that when all known observables have acknowledged a message it can be removed.
+    #[serde(rename = "interest")]
+    Interest = 1,
+    /// `WorkQueue` specifies that when the first worker or subscriber acknowledges the message it can be removed.
+    #[serde(rename = "workqueue")]
+    WorkQueue = 2,
+}
+
+impl Default for RetentionPolicy {
+    fn default() -> RetentionPolicy {
+        RetentionPolicy::Limits
+    }
+}
+
+/// determines how messages are stored for retention.
+#[derive(Debug, Serialize, Deserialize, Clone, Copy, PartialEq, Eq)]
+#[repr(u8)]
+pub enum StorageType {
+    /// Stream data is kept in files. This is the default.
+    #[serde(rename = "file")]
+    File = 0,
+    /// Stream data is kept only in memory.
+    #[serde(rename = "memory")]
+    Memory = 1,
+}
+
+impl Default for StorageType {
+    fn default() -> StorageType {
+        StorageType::File
+    }
+}
+/// A UTC time
+pub type DateTime = time::OffsetDateTime;
+
+/// Shows config and current state for this stream.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct StreamInfo {
+    /// The configuration associated with this stream
+    pub config: StreamConfig,
+    /// The time that this stream was created
+    #[serde(with = "rfc3339")]
+    pub created: DateTime,
+    /// Various metrics associated with this stream
+    pub state: StreamState,
+}
+
+/// information about the given stream.
+#[derive(Debug, Serialize, Deserialize, Clone, Copy)]
+pub struct StreamState {
+    /// The number of messages contained in this stream
+    pub messages: u64,
+    /// The number of bytes of all messages contained in this stream
+    pub bytes: u64,
+    /// The lowest sequence number still present in this stream
+    pub first_sequence: u64,
+    /// The time associated with the oldest message still present in this stream
+    #[serde(with = "rfc3339")]
+    pub first_timestamp: DateTime,
+    /// The last sequence number assigned to a message in this stream
+    pub last_sequence: u64,
```

Comment:
Oh btw you're gonna have to serde rename these renames.

```suggestion
    #[serde(rename="last_seq")]
    pub last_sequence: u64,
```